internal/extsvc/bitbucketserver/client.go

```diff
@@ -986,7 +986,11 @@ func (c *Client) do(ctx context.Context, req *http.Request, result any) (*http.R
 	if s, ok := result.(*[]byte); ok {
 		*s = bs
 	} else if result != nil {
-		return resp, json.Unmarshal(bs, result)
+		err := json.Unmarshal(bs, result)
+		if err != nil {
+			return resp, errors.Wrap(err, "failed to unmarshal response to JSON")
+		}
+		return resp, nil
```

Comment:
you can even do:
```suggestion
		return resp, errors.Wrap(json.Unmarshal(bs, result), "failed to unmarshal response to JSON")
```

when an error from unmarshal is `nil` -- it is still `nil` after wrapping

Reply:
Aha yes! Great catch - I forgot about that.